File: 프로그래머스/lv2/17687. ［3차］ n진수 게임/［3차］ n진수 게임.js
function solution(n, t, m, p) {
    const max = m * (t - 1) + p;
    const answer = [];
    let numbers = "";
    let value = 0;

    while (numbers.length < max) numbers += (value++).toString(n);

    let i = 0;
    while (answer.length < t) {
        if ((i % m) + 1 === p) answer.push(numbers[i]);
        i++;
    }
    return answer.join("").toUpperCase();
}

// 제너레이터 활용
function solution(n, t, m, p) {
    const program = (function* (n) {
        let value = 0;
        while (true) {
            const numbers = [...value.toString(n)];
            for (let i = 0; i < numbers.length; i++)
                yield numbers[i].toString().toUpperCase();
            value++;
        }
    })(n);

    const output = [];
    let i = 0;
    while (output.length < t) {
        const { value } = program.next();
        if (i++ % m === p - 1) output.push(value);
    }
    return output.join("");
}

File: 프로그래머스/lv2/17687. ［3차］ n진수 게임/［3차］ n진수 게임.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

string to_bit(int num, const int& bit) {
    string ans = "";
    string hex[] = { "A", "B", "C", "D", "E", "F" };

    while (num > 0) {
        const auto& val = num % bit;
        ans += (val >= 10) ? hex[val - 10] : to_string(val);
        num /= bit;
    }
    reverse(ans.begin(), ans.end());
    return ans;
}

string solution(int n, int t, int m, int p) {
    string answer = "";
    string list = "0";
    
    int j = 1;
    while (list.length() < t * m)
        list += (to_bit(j++, n));

    j = 0;
    while (answer.length() < t)
        answer += list[(p - 1) + (m * j++)];

    return answer;
}